samples/boards/stm32/usbc/sink/src/usbc_snk.c

```diff
@@ -0,0 +1,775 @@
+	SUB_STATE0,
+	/** Generic substate 1 */
+	SUB_STATE1,
+	/** Generic substate 2 */
+	SUB_STATE2,
+	/** Generic substate 3 */
+	SUB_STATE3,
+};
+
+/* TypeC Only Power strings (No PD) */
+static const char *const pwr2_5_str = "5V/0.5A";
+static const char *const pwr7_5_str = "5V/1.5A";
+static const char *const pwr15_str = "5V/3A";
+static const char *const pwr_open_str = "Removed";
+
+/**
+ * @brief TypeC PD object
+ */
+static struct tc_t {
+	/** state machine context */
+	struct smf_ctx ctx;
+	/** TypeC Port Controller device */
+	const struct device *tcpc_dev;
+	/** VBUS measurement device */
+	const struct device *vbus_dev;
+	/** Port polarity */
+	enum tc_cc_polarity cc_polarity;
+	/** Time a port shall wait before it can determine it is attached */
+	uint32_t cc_debounce;
+	/** The cc state */
+	enum tc_cc_states cc_state;
+	/** Voltage on CC pin */
+	enum tc_cc_voltage_state cc_voltage;
+	/** Current CC1 value */
+	enum tc_cc_voltage_state cc1;
+	/** Current CC2 value */
+	enum tc_cc_voltage_state cc2;
+	/** PD message for RX and TX */
+	struct pd_msg msg;
+	/** PD connection flag. Also used as PD contract in place flag */
+	bool pd_connected;
+	/** PD state */
+	enum pd_states pd_state;
+	/** Event flags */
+	uint32_t flag;
+	/** Transmission message id count */
+	uint32_t msg_id_cnt;
+	/** Receive message id */
+	uint32_t msg_id;
+	/** General timer */
+	uint32_t timer;
+	/** Sub state variable */
+	enum substate sub_state;
+	/** Source caps */
+	uint32_t src_caps[7];
+	/** Source caps header */
+	union pd_header src_caps_hdr;
+} tc;
+
+static void set_state_tc(const enum usb_tc_state new_state);
+
+/**
+ * @brief Sink startup
+ */
+static void snk_startup(struct tc_t *tc)
+{
+	tc->flag = 0;
+	tc->msg_id_cnt = 0;
+	tc->msg_id = 0;
+	tc->timer = 0;
+
+	/* Not currently PD connected */
+	tc->pd_connected = false;
+
+	/* Initialize PD state */
+	tc->pd_state = SNK_WAIT_FOR_CAPABILITIES;
+}
+
+/**
+ * @brief Sink power sub states. Only called if a PD contract is ot in place
+ */
+static void sink_power_sub_states(void)
+{
+	enum tc_cc_voltage_state cc;
+	enum tc_cc_voltage_state new_cc_voltage;
+	char const *pwr;
+
+	cc = tc.cc_polarity ? tc.cc2 : tc.cc1;
+
+	if (cc == TC_CC_VOLT_RP_DEF) {
+		new_cc_voltage = TC_CC_VOLT_RP_DEF;
+		pwr = pwr2_5_str;
+	} else if (cc == TC_CC_VOLT_RP_1A5) {
+		new_cc_voltage = TC_CC_VOLT_RP_1A5;
+		pwr = pwr7_5_str;
+	} else if (cc == TC_CC_VOLT_RP_3A0) {
+		new_cc_voltage = TC_CC_VOLT_RP_3A0;
+		pwr = pwr15_str;
+	} else {
+		new_cc_voltage = TC_CC_VOLT_OPEN;
+		pwr = pwr_open_str;
+	}
+
+	/* Debounce the cc state */
+	if (new_cc_voltage != tc.cc_voltage) {
+		tc.cc_voltage = new_cc_voltage;
+		tc.cc_debounce = T_RP_VALUE_CHANGE;
+	} else if (tc.cc_debounce) {
+		/* Update timer */
+		tc.cc_debounce--;
+	}
+}
+
+/*
+ * TYPE-C State Implementations
+ */
+
+/**
+ * @brief Unattached.SNK
+ */
+static void tc_unattached_snk_entry(void *tc_obj)
+{
+	struct tc_t *tc = tc_obj;
+
+	printk("UnAttached.SNK\n");
```

Comment:
Better to use either LOG_* or printk, since it looks like it is only there for logging purposes I suggest to use LOG_ instead of printk in `usbc_snk.c`

Reply:
I've changed the prinks to LOG_INF